From 179516b0b7b3b4a0920444288cf8acca872e8846 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Bart=C5=82omiej=20Piotrowski?= <b@bpiotrowski.pl>
Date: Mon, 21 Dec 2015 19:49:51 +0100
Subject: [PATCH] Explicitly define namespace of ifstream and ofstream

With GCC 5.3.0 and Boost 1.60.0, the build fails with following error:
/build/libasl/src/ASL-0.1.6/src/utilities/aslParametersManager.cxx:466:6:
error: reference to 'ifstream' is ambiguous
      ifstream ifs(p.string());
      ^
/usr/include/boost/filesystem/fstream.hpp:169:32: note: candidates are:
typedef class boost::filesystem::basic_ifstream<char> boost::filesystem::ifstream
   typedef basic_ifstream<char> ifstream;
                                ^
---
 src/utilities/aslParametersManager.cxx | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/utilities/aslParametersManager.cxx b/src/utilities/aslParametersManager.cxx
index 98a0192..d6ef3c7 100644
--- a/src/utilities/aslParametersManager.cxx
+++ b/src/utilities/aslParametersManager.cxx
@@ -332,7 +332,7 @@ namespace asl
 
 		try
 		{
-			ifstream ifs(paramFile);
+			std::ifstream ifs(paramFile);
 			if (!ifs.good())
 				errorMessage("Can not open parameters file: " + paramFile);
 
@@ -357,7 +357,7 @@ namespace asl
 
 	void ParametersManager::writeParametersFile(const std::string fileName)
 	{
-		ofstream fo(fileName);
+		std::ofstream fo(fileName);
 		if (!fo.good())
 			errorMessage("ParametersManager::writeParametersFile() - can not open file: " + fileName);
 
@@ -463,7 +463,7 @@ namespace asl
 				}
 				else
 				{
-					ifstream ifs(p.string());
+					std::ifstream ifs(p.string());
 					if (ifs.good())
 					{
 						parsed_options parsed = parse_config_file(ifs, allOptions, true);
@@ -514,4 +514,4 @@ namespace asl
 		}
 	}
 		 
-} //namespace asl
\ No newline at end of file
+} //namespace asl
-- 
2.6.4

